drivers/led_strip/tlc59731.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2024 Javad Rahimipetroudi <javad.rahimipetroudi@mind.be>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tlc59731
+
+/**
+ * @file
+ * @brief LED driver for the TLC59731 LED driver.
+ *
+ * TLC59731 is a 3-Channel, 8-Bit, PWM LED Driver
+ * With Single-Wire Interface (EasySet)
+ *
+ */
+
+#include <zephyr/drivers/led_strip.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(tlc59731, CONFIG_LED_STRIP_LOG_LEVEL);
+
+/* Pulse timing */
+#define DELAY     0x01 /* us */
+#define T_CYCLE_0 0x04 /* us */
```

Comment:
Consider adding a prefix `TLC59731_` or `EASYJET_` to every definition to avoid possible redefinition(s).